src/ripple/rpc/handlers/Tx.cpp

```diff
@@ -245,6 +264,15 @@ doTxHelp(RPC::Context& context, TxArgs const& args)
         }
         result.validated = isValidated(
             context.ledgerMaster, ledger->info().seq, ledger->info().hash);
+
+        // compute outgoing CTID
+        uint32_t lgrSeq = ledger->info().seq;
+        uint32_t txnIdx = meta->getAsObject().getFieldU32(sfTransactionIndex);
+        uint32_t netID = context.app.config().NETWORK_ID;
+
+        if (txnIdx <= 0xFFFFU && netID < 0xFFFFU && lgrSeq < 0x0FFF'FFFFUL)
+            result.ctid =
+                RPC::encodeCTID(lgrSeq, (uint16_t)txnIdx, (uint16_t)netID);
```

Comment:
I'd recommend putting these checks in a single place (namely in `encodeCTID`) so that you don't need to check in "unrelated" places and keep the code cleaner.